doc/lua_api.txt

```diff
@@ -2281,6 +2360,41 @@ Elements
 **Note**: do _not_ use a element name starting with `key_`; those names are
 reserved to pass key press events to formspec!
 
+Migrating to Real Coordinates
+-----------------------------
+
+In the old system, positions included padding and spacing. Padding is a gap between
+the formspec window edges and content, and spacing is the gaps between items. For
+example, two `1x1` elements at `0,0` and `1,1` would have a spacing of `5/4` between them,
+and a padding of `3/8` from the formspec edge. It may be easiest to recreate old layouts
+in the new coordinate system from scratch.
+
+To recreate an old layout with padding, you'll need to pass the positions and sizes
+through the following formula to re-introduce padding:
+
+```
+pos = (oldpos + 1)*spacing + padding
+where
+    padding = 3/8
+    spacing = 5/4
+```
+
+You'll need to change the `size[]` tag like this:
+
+```
+size = (oldsize-1)*spacing + padding*2 + 1
+```
+
+A few elements had random offsets in the old system. Here is a table which shows these
+offsets when migrating:
+
+| Element |  Position  |  Size   | Notes
+|---------|------------|---------|-------
+| box     | +0.3, +0.1 | 0, -0.4 |
+| button  |            |         | Buttons now support height, so you will need to set h=1 and reposition if h!=1 before
+| list    |            |         | Note: spacing is now 0.25 for both directions, meaning lists will be taller in height
```

Comment:
- `Note:` in `Notes` is redundant.
- The lines are a bit long (also in the other lines).

Maybe do something like this?
```suggestion
| list    |            |         | spacing is now 0.25 for both directions, meaning lists
|         |            |         | will be taller in height
```

Reply:
You can't place it on two lines like that because it will add another row to the table, like so:

```
| Element |  Position  |  Size   | Notes
|---------|------------|---------|-------
| list    |            |         | spacing is now 0.25 for both directions, meaning
|         |            |         | lists will be taller in height
```

| Element |  Position  |  Size   | Notes
|---------|------------|---------|-------
| list    |            |         | spacing is now 0.25 for both directions, meaning
|         |            |         | lists will be taller in height

Even if you add a plain newline without pipes, it still doesn't work:

```
| Element |  Position  |  Size   | Notes
|---------|------------|---------|-------
| list    |            |         | spacing is now 0.25 for both directions, meaning
lists will be taller in height
| bla     |            |         |
```
| Element |  Position  |  Size   | Notes
|---------|------------|---------|-------
| list    |            |         | spacing is now 0.25 for both directions, meaning
lists will be taller in height
| bla     |            |         |

Markdown really didn't plan for this.  I don't think we should change this to have improper markdown.

Although yes, `Note:` is redundant.  I just copied and pasted what Rubenwardy wrote (with a single correction).  I'll fix that.

Reply:
Oh, I forgot that `lua_api.txt` is written in a way that it could be a markdown file.
You are right.